### PR TITLE
tests: net: Fix possible null dereference in net_pkt tests

### LINKED ISSUE #10777: tests/net/net_pkt/src/main.c : NULL dereferences
Found using Coccinelle.
https://github.com/zephyrproject-rtos/zephyr/pull/10776
```
himanshu@himanshu-Vostro-3559:~/zephyr$ ./scripts/coccicheck --mode=report --cocci=scripts/coccinelle/deref_null.cocci --verbose=1 tests/

Please check for false positives in the output before submitting a patch.
When using "patch" mode, carefully review the patch before submitting it.

Processing deref_null.cocci
with option(s) ""

Message example to submit a patch:

 A variable is dereferenced under a NULL test.
 Even though it is known to be NULL.

 The semantic patch that makes this report is available
 in scripts/coccinelle/deref_null.cocci.

 More information about semantic patching is available at
 http://coccinelle.lip6.fr/

Running (4 in parallel): /usr/local/bin/spatch -D report --no-show-diff --very-quiet --cocci-file scripts/coccinelle/deref_null.cocci --patch /home/himanshu/zephyr --dir tests/ --jobs 4 --chunksize 1
tests/net/net_pkt/src/main.c:448:16-19: ERROR: tfrag is NULL but dereferenced.
tests/net/net_pkt/src/main.c:448:33-36: ERROR: tfrag is NULL but dereferenced.
```

### PR DESCRIPTION
**Pull request by jukkar**
It might be that the test might access null pointer in some cases.
In practice this is not possible as the test is using pre-defined
data when running the test.

Fixes #10777

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

### CODE REVIEW

File: tests/net/net_pkt/prj.conf [hunk 1]
```diff
@@ -15,3 +15,7 @@ CONFIG_NET_LOG=y
 CONFIG_ENTROPY_GENERATOR=y
 CONFIG_TEST_RANDOM_GENERATOR=y
 CONFIG_ZTEST=y
+
+# Do not print any errors, they just confuse the sanitychecker
+# even if there is no problem with the test.
+CONFIG_NET_MAX_LOG_LEVEL_OFF=y
```
Comment (pfalcon) [inline tests/net/net_pkt/prj.conf]: I'm not sure how this is related to mentioned "Fixes #10777".

Comment (pfalcon) [reply tests/net/net_pkt/prj.conf]: (Or the commit message for that matter either.)

Comment (jukkar) [reply tests/net/net_pkt/prj.conf]: The test was printing some errors that were not related to the passing of the test. That is the reasong for the setting.

Comment (pfalcon) [reply tests/net/net_pkt/prj.conf]: What if later it turns out that errors were useful/interesting, but nobody will have a chance to see them?

P.S. If commit message have a sample of kind of messages which are disabled, there wouldn't be these questions.

**Comment by jukkar** [comment]
I removed the max log level setting from new version as that seemed to cause too much confusion.

**Comment by pfalcon** [comment]
> I removed the max log level setting from new version as that seemed to cause too much confusion.

Not that much, just a bit ;-).

**Review by pfalcon — APPROVED** [review]

**Pull request merged**